spec/src/main/resources/META-INF/smithy/bsp/bsp.smithy

```diff
@@ -950,6 +963,42 @@ structure InverseSourcesResult {
     targets: BuildTargetIdentifiers
 }
 
+structure WrappedSourceItem {
+    @required
+    uri: String
+    @required
+    generatedUri: String
+    @required
+    topWrapper: String
+    @required
+    bottomWrapper: String
+}
```

Comment:
Do we really need to add a new request for this? Or should we consider "wrapped sources" as part of `buildTarget/sources` ?

I would rather go with the second option, and add the optional `topWrapper`, `bottomWrapper` to the `SourceItem`:

```smithy
structure SourceItem {
  @required  
  uri: URI;
  @required   
  kind: SourceItemKind;
  @required  
  generated: boolean;
  // optional
  topWrapper: String;
  // optional
  bottomWrapper: String
}
```

Or even better:
```smithy
structure SourceItem {
  @required  
  uri: URI;
  @required   
  kind: SourceItemKind;
  @required  
  generated: boolean;
  // optional
  wrapper: SourceWrapper;
}

structure SourceWrapper {
  @required  
  prefix: String;
  @required   
  suffix: String;
}
```

Reply:
Oh yes I missed that part.

So it seems a "wrapped source" is the source of a generated sources. And we want to have the support of our IDE in it because it looks a lot like a Scala file.

But this seems to be a special case of something more general which are the source files of source generation. In this case it is an sc file that generates a Scala file, but it could also be a proto, json, smithy or twirl file that generates Scala code.

So can we generalize that request? Something like this maybe:

```smithy
@jsonRequest("buildTarget/sourceGenerationSources")
operation BuildTargetSourceGenerationSources {
    input: SourceGenerationSourcesParams
    output: SourceGenerationSourcesResult
}

structure SourceGenerationSourcesItem {
  @required
  uri: String
  @required
  generatedUri: String
  // optional
  data: SourceGenerationSourcesItemData
}

// in the scala extension
@dataKind(kind: "wrapped-scala-source", extends: [SourceGenerationSourcesItemData])
structure WrappedScalaSource {
    @required
    prefix: String
   @required
    suffix: String
}
```